src/compiler.ts

```diff
@@ -462,6 +462,36 @@ export class Compiler extends DiagnosticEmitter {
     return module;
   }
 
+  addNullPointerExceptionHandling(module: Module) {
+    if (this.options.hasFeature(Feature.EXCEPTION_HANDLING)) {
+      let message = this.ensureStaticString("null is not a function");
+      let exceptionType = module.addFunctionType(
+        "NullPointerException",
+        NativeType.None,
+        [NativeType.I32]
+      );
+
+      module.addEvent("NullPointerException", 0, exceptionType);
+      module.addFunction(
+        "null",
+        this.ensureFunctionType(null, Type.void),
+        null,
+        module.block(null, [
+          module.throw("NullPointerException", [module.i32(message)])
```

Comment:
Here, `message` already is an `i32.const`, so this makes an `i32.const(someRandomPtrToBinaryenMemory)`. Wondering if this is related to the string not showing up in static data.

Reply:
Nice catch! That's indeed a bug, although, it didn't seem to solve the issue when I regenerate the files. Only change was:

```diff
-  i32.const 5460624
+  i32.const 24
```

Which that number makes much more sense for a data segment index (5460624 in hindsight should have been an obvious give away that something was wrong)

Reply:
Ohh I think I found the issue. I think it's too late, the module.setMemory call is made _above_, before the call to addNullPointerExceptionHandling, so the module misses out on any changes to the internal this.memorySegments storage.

Checking...

Reply:
Yep that was the issue.